src/components/calculator.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import calculate from './logic/calculate';
+
+class CalcUserInterface extends React.Component { // eslint-disable-line
+  constructor(props) {
+    super(props);
+    this.state = { total: 0, next: null, operation: null };
+  }
+
+  handleClick = (event) => {
+    const buttonName = event.target.innerText;
+    const result = calculate(this.state, buttonName);
+    this.setState(result);
+  }
+
+  render() {
+    const { total, next, operation } = this.state;
+    return (
```

Comment:
- [ ] Please pass the props correctly from the `Calculator` component to its child component and validate those props using propTypes.
- [ ] Also, please make use of the known react hook `useState` to store and mutate the state
_This is as per project requirements. see [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_events.md#:~:text=Note%3A%20Practice%20what%20you%20have%20learned%20until%20now%3A%20use%20useState%20and%20props.)._

Reply:
resolved successively